components/script/dom/websocket.rs

```diff
@@ -168,19 +166,12 @@ impl WebSocket {
         if !self.clearing_buffer.get() && self.ready_state.get() == WebSocketRequestState::Open {
             self.clearing_buffer.set(true);
 
-            let task = Box::new(BufferedAmountTask { address });
-
-            let pipeline_id = self.global().pipeline_id();
-            self.global()
-                .script_chan()
-                // TODO: Use a dedicated `websocket-task-source` task source instead.
-                .send(CommonScriptMsg::Task(
-                    WebSocketEvent,
-                    task,
-                    Some(pipeline_id),
-                    WebsocketTaskSource::NAME,
-                ))
-                .unwrap();
+            // TODO(mrobinson): Should this task be cancellable?
+            let _ = self
+                .global()
+                .task_manager()
+                .websocket_task_source()
+                .queue_unconditionally(BufferedAmountTask { address });
```

Comment:
Doesn't this change here to use `queue_unconditionally` instead of manually sending the `CommonScriptMsg` mean the Task's `ScriptThreadEventCategory` is now `NetworkEvent` instead of `WebSocketEvent` ? Is that fine? Based on a quick check, now the `WebSocketEvent` is not used anywhere else (except in type functions)

Reply:
Oh, interesting. I think the right thing here is that the WebSocket task source should be classifying things as `ScriptThreadEventCategory::WebSocket`. I've made this change. I imagine that using `NetworkingEvent` was an oversight before.